### PR TITLE
Fixed MSVC solution to compile straight out of the box

### PR DESCRIPTION
**Pull request by D-POWER**
I noticed that the MSVC solution was not correctly setup. Some projects generate binaries in the same directory regardless of the current build target. And I fixed a few Windows-specific compilation errors.

### CODE REVIEW

File: msvc/test-metadata.vcxproj [hunk 1]
```diff
@@ -117,6 +117,7 @@
       </PrecompiledHeader>
       <WarningLevel>Level3</WarningLevel>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
+      <AdditionalIncludeDirectories>..\;..\eglib\src;C:\cygwin\opt\mono\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (kumpera) [inline msvc/test-metadata.vcxproj]: We don't want hardcoded paths in the build scripts

**Comment by D-POWER** [comment]
No it's not needed. The folder *opt* points to a directory that contains Mono include folder, but we can replace that with a post-build event in Visual Studio which copies all necessary Mono headers to a local directory.

**Comment by kumpera** [comment]
Hey @tritao could you review this one?

**Comment by tritao** [comment]
Hey @D-POWER, seems like your change to to link library dependencies, instead of explicitly passing the libs breaks the SGen build configurations. This is because libgc is still a dependent project and it tries to look for a `libgc.lib` which does not exist.

**Pull request merged**